beater/config/config.go

```diff
@@ -92,6 +100,59 @@ type Config struct {
 	DefaultServiceEnvironment string                  `config:"default_service_environment"`
 
 	Pipeline string
+
+	// TODO: wat is with this naming
+	AgentConfigs []ServiceConfig `config:"agent_config"`
+}
+
+// ServiceConfig defines configuration for agents.
+type ServiceConfig struct {
```

Comment:
There's already an AgentConfig struct defined, any preference on how to navigate that?

https://github.com/elastic/apm-server/blob/master/beater/config/config.go#L85

Reply:
Ah bugger :( It may be confusing that there's both "agent_config" and "agent.config" in `apm-server.yml`. I'm not sure how to fix this. Current best thought is to rename AgentConfig to KibanaAgentConfig.